### PR TITLE
Ffmpeg: build both shared and static libs

### PR DESCRIPTION
**Pull request by ImperatorS79**
The patch directly comes from here: http://git.videolan.org/?p=ffmpeg.git;a=commitdiff;h=7c59e1b0f285cd7c7b35fcd71f49c5fd52cf9315. It seems srt was updated and removed some deprecated constant.

I removed the pgp key check because it did not work on my end but I can add it again if someone explains me how it works ^^.

EDIT: without the patch it did not compile locally

### CODE REVIEW

**Comment by Biswa96** [comment]
May I summon a VLC developer just to verify? cc @robUx4

**Comment by ImperatorS79** [comment]
The patch comes from here: https://trac.ffmpeg.org/ticket/8760

**Comment by robUx4** [comment]
The addition seems mostly about adding the shared version but we don't use that in VLC. We don't use SRT in ffmpeg either but as a standalone module. So I'm not sure I can be of much help here.

File: mingw-w64-ffmpeg/PKGBUILD [hunk 1]
```diff
@@ -47,18 +47,19 @@ depends=("${MINGW_PACKAGE_PREFIX}-aom"
 makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "${MINGW_PACKAGE_PREFIX}-nasm")
-source=(https://ffmpeg.org/releases/${_realname}-${pkgver}.tar.xz{,.asc})
-validpgpkeys=('FCF986EA15E6E293A5644F10B4322F04D67658D8')
+source=(https://ffmpeg.org/releases/${_realname}-${pkgver}.tar.xz
+        srt.patch)
```
Comment (lazka) [inline mingw-w64-ffmpeg/PKGBUILD:51]: I've moved this to #7336

File: mingw-w64-ffmpeg/PKGBUILD [hunk 2]
```diff
@@ -47,18 +47,19 @@ depends=("${MINGW_PACKAGE_PREFIX}-aom"
 makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "${MINGW_PACKAGE_PREFIX}-nasm")
-source=(https://ffmpeg.org/releases/${_realname}-${pkgver}.tar.xz{,.asc})
-validpgpkeys=('FCF986EA15E6E293A5644F10B4322F04D67658D8')
```
Comment (lazka) [inline mingw-w64-ffmpeg/PKGBUILD:51]: You can use `--skippgpcheck` if you don't want to verify the signature. Please don't delete this.

File: mingw-w64-ffmpeg/PKGBUILD [hunk 3]
```diff
@@ -121,6 +120,69 @@ build() {
...
+  mkdir -p "${srcdir}/build-${MINGW_CHOST}-shared" && cd "${srcdir}/build-${MINGW_CHOST}-shared"
+
+  # Fix using SRT headers
+  CFLAGS+=" -DWIN32"
+  CXXFLAGS+=" -DWIN32"
+
+  CC=${MINGW_PREFIX}/bin/gcc \
+  CXX=${MINGW_PREFIX}/bin/g++ \
+  ../${_realname}-${pkgver}/configure \
+    --prefix=${MINGW_PREFIX} \
```
Comment (lazka) [inline mingw-w64-ffmpeg/PKGBUILD:134]: Not a fan of duplicating all those build flags. Can we share them between the builds?

File: mingw-w64-ffmpeg/PKGBUILD [hunk 4]
```diff
@@ -121,6 +120,69 @@ build() {
       exit 1
     }
   make
+  
+  [[ -d "${srcdir}/build-${MINGW_CHOST}-shared" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}-shared"
+  mkdir -p "${srcdir}/build-${MINGW_CHOST}-shared" && cd "${srcdir}/build-${MINGW_CHOST}-shared"
+
+  # Fix using SRT headers
+  CFLAGS+=" -DWIN32"
```
Comment (lazka) [inline mingw-w64-ffmpeg/PKGBUILD:128]: Why is this needed again?

File: mingw-w64-ffmpeg/PKGBUILD (hunk 2 above)
Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:51]: Okay thanks for the info

File: mingw-w64-ffmpeg/PKGBUILD (hunk 1 above)
Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:51]: I'll merge master into my branch once it is committed.

File: mingw-w64-ffmpeg/PKGBUILD (hunk 4 above)
Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:128]: Great question but no idea ^^.

File: mingw-w64-ffmpeg/PKGBUILD (hunk 3 above)
Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:136]: I do not think so. You need to configure each static and shared build separetly.

File: mingw-w64-ffmpeg/PKGBUILD (hunk 4 above)
Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:130]: Oh, I did not see I copy-pasted this twice! Sorry ^^'.

File: mingw-w64-ffmpeg/PKGBUILD (hunk 3 above)
Comment (lazka) [reply mingw-w64-ffmpeg/PKGBUILD:132]: I mean the configure options, not the call to configure itself

Comment (ImperatorS79) [reply mingw-w64-ffmpeg/PKGBUILD:132]: Yes, I understood that too. But I do not of a way to do that in bash.

**Comment by lazka** [comment]
@ImperatorS79 I've pushed something. Does it look OK to you?

**Comment by ImperatorS79** [comment]
Since it compiles it looks good ^^.

**Comment by lazka** [comment]
if that's enough for you, ok :) we don't use static libs, so don't expect any testing in that area.
